x/gamm/pool-models/stableswap/pool.go

```diff
@@ -69,6 +78,45 @@ func (pa Pool) getPoolAmts(denoms ...string) ([]sdk.Int, error) {
 	return result, nil
 }
 
+// getScaledPoolAmts returns scaled amount of pool liquidity based on each asset's precisions
+func (pa Pool) getScaledPoolAmts(denoms ...string) ([]sdk.Int, error) {
+	result := make([]sdk.Int, len(denoms))
+	poolLiquidity := pa.PoolLiquidity
+	liquidityIndexes := pa.getLiquidityIndexMap(poolLiquidity)
+
+	for i, denom := range denoms {
+		liquidityIndex := liquidityIndexes[denom]
+
+		amt := poolLiquidity.AmountOf(denom)
+		if amt.IsZero() {
+			return []sdk.Int{}, fmt.Errorf("denom %s does not exist in pool", denom)
+		}
+		scalingFactor := pa.GetScalingFactorByLiquidityIndex(liquidityIndex)
+		result[i] = amt.QuoRaw(int64(scalingFactor))
+	}
+	return result, nil
+}
+
+// getDescaledPoolAmts gets descaled amount of given denom and amount
+func (pa Pool) getDescaledPoolAmt(denom string, amount sdk.Dec) sdk.Dec {
+	poolLiquidity := pa.PoolLiquidity
+
+	liquidityIndexes := pa.getLiquidityIndexMap(poolLiquidity)
```

Comment:
Actually, why do we even need to pass it as an argument? Since `getLiquidityIndexMap` is method on the pool, we should be able to access it from the receiver